### ISSUE TITLE
Bug: Trying to access data with reduct-py can lead to different errors, or successful download

### ISSUE THREAD

**Issue by mmaciejak** [reductstore/reduct-py#107]
### What happened?

While trying to get data from reduct store over vpn connection, the operation can either end successfully or multiple different errors can occur for the same time range. Reduct store is set up by @toolkiz and he never encountered this issue while running the download script locally.  

### Errors I got so far

1.  `reduct.error.ReductError: Status 404: Query 1 not found`
2.  `ValueError: buffer size must be a multiple of element size`
3.  `ValueError: cannot reshape array of size 62 into shape (3)` (the size of the array also appears to change on each query)
4.  `ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (3374,) + inhomogeneous part.` in this case the download is sucesfull but the data is corrupted, from what I checked the packages appear "splitted" 
```
(24, 3)
(24, 3)
(24, 3)
(17, 3)
(7, 3)
(24, 3)
(24, 3)
(24, 3)
(24, 3)
```

### Version

reduct-py==1.8.1

### OS

Linux

### Snippet to repeat the bug

```
async def get_array_from_reduct(bucket_name, start_ts, stop_ts):
    # Create a ReductStore client
    print(f"getting data from reduct for range {start_ts} to {stop_ts}")
    async with Client("http://100.96.1.13:8383") as client:
        
        # creating the bucket
        bucket = await client.create_bucket(
            "likora",
            BucketSettings(quota_type=QuotaType.NONE),
            exist_ok=True,
        )
        
        all_data_within_time = []
        async for record in bucket.query(bucket_name, start=start_ts*1000000, stop=stop_ts*1000000):
            async for data in record.read(record.size):
                
                # data processing 
                # ------------------
                blender_data = np.frombuffer(data, dtype=record.labels['dtype'])
                blender_data = blender_data.reshape(-1, int(record.labels['parameters']))

                all_data_within_time.append(blender_data)
        
        data_array = np.array(all_data_within_time)
        print(f"Got array of size {data_array.shape}")
        return data_array
```
`bucket_name` is actually an entry name, which I know is confusing, sorry for that. 

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by atimin** [comment]
Hi @mmaciejak, as we instigated the issues together,  the call `async for data in record.read(record.size):` doesn't work correctly. Since this is a bug in the Python SDK, I moved the issue to the `reduct-py` repository and start working on a fix. You will see when I've done. The fix will be delivered in the 1.9.0 version of the SDK.

**Comment by atimin** [comment]
@mmaciejak , this isn't a bug, actually. The `record.read(n:int) call is using the `iter_chuncked` function from aiohttp:

 https://docs.aiohttp.org/en/stable/streams.html#aiohttp.StreamReader.iter_chunked

As you can see, the size of a chunk is a maximum possible size. It means that your loop may need a few iterations to download the whole record. So you should re-write your loop to make it robust:

```py
        all_data_within_time = []
        async for record in bucket.query(bucket_name, start=start_ts*1000000, stop=stop_ts*1000000):
            data = b""
            async for chunk in record.read(record.size):
                data += chunk

            # data processing 
            # ------------------
            blender_data = np.frombuffer(data, dtype=record.labels['dtype'])
            blender_data = blender_data.reshape(-1, int(record.labels['parameters']))

            all_data_within_time.append(blender_data)
```

I'll update documentation and close the bug.
